albogdan/skule-vote
# Updated Models and Migration

## Overview

- Resolves #43
- Each `Election` should only have at most one `Eligibility`
- Changed helper text of Rule Violation (RV) link to be consistent with Disqualified (DQ) link: "(Optional) Enter a link to the rule violation ruling."
- Changed DQ helper text blurb to "(Optional) Enter a disqualification ruling message for this Candidate."
- Renamed `preamble` to `statement` 
- Removed `blurb`
- Added election_id column to the Eligibility table on the admin site
- Changed `first_year_eligible`, `second_year_eligible` fields to `year_1_eligible`, `year_2_eligible`, etc.

## Unit Tests Created

- N/A


## Steps to QA

- First once you pull do the following to apply the migration:
   - `python manage.py migrate backend zero`
   - `python manage.py migrate backend`
- Go to the admin site and see if all the changes are as expected
- Go to the admin site and create an `ElectionSession`, `Election`, and an `Eligibility` for that `Election`. Now try to create another `Eligibility` for that same `Election`. You should get an error on the form that says `Eligibility with this Election already exists.` in red.
- Go to the `Eligibility` list view. You should see an `election_id` beside each `Eligibility` you created
